pkg/cnservice/tae.go

```diff
@@ -40,7 +45,29 @@ func initTAE(
 	}
 	syscall.Umask(mask)
 
-	tae, err := db.Open(targetDir+"/tae", nil)
+	opts := &options.Options{}
+	switch cfg.Engine.Logstore {
+	case options.LogstoreLogservice:
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		lc, err := logservice.NewClient(ctx, logservice.ClientConfig{
+			ReadOnly:         false,
+			LogShardID:       1,
+			DNReplicaID:      1,
+			ServiceAddresses: cfg.HAKeeper.ClientConfig.ServiceAddresses,
+		})
+		cancel()
+		if err != nil {
+			panic(err)
+		}
+		opts.Lc = lc
+		opts.LogStoreT = options.LogstoreLogservice
+	case options.LogstoreBatchStore, "":
+		opts.LogStoreT = options.LogstoreBatchStore
+	default:
+		panic(fmt.Sprintf("invalid logstore type: %v", cfg.Engine.Logstore))
+	}
+
+	tae, err := db.Open(targetDir+"/tae", opts)
 	if err != nil {
 		logutil.Infof("Open tae failed. error:%v", err)
```

Comment:
Do you want to return err or panic?   Either way, but better be consistent.

Reply:
OK, I'll return moerr.